src/main/java/net/glowstone/entity/FishingRewardManager.java

```diff
@@ -0,0 +1,143 @@
+package net.glowstone.entity;
+
+import com.google.common.collect.Multimap;
+import com.google.common.collect.MultimapBuilder;
+import java.io.InputStreamReader;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.configuration.serialization.ConfigurationSerializable;
+import org.bukkit.inventory.ItemStack;
+
+/**
+ * Generates fishing loot.
+ */
+public class FishingRewardManager {
+
+    private final Multimap<RewardCategory, RewardItem> values = MultimapBuilder
+            .enumKeys(RewardCategory.class).arrayListValues().build();
+
+    /**
+     * Creates the instance.
+     */
+    public FishingRewardManager() {
+        YamlConfiguration builtinValues = YamlConfiguration.loadConfiguration(
+                new InputStreamReader(getClass().getClassLoader()
+                        .getResourceAsStream("builtin/fishingRewards.yml")));
+
+        registerBuiltins(builtinValues);
+    }
+
+    private void registerBuiltins(ConfigurationSection mainSection) {
+        ConfigurationSection valuesSection = mainSection.getConfigurationSection("rewards");
+        Set<String> categories = valuesSection.getKeys(false);
+        for (String strCategorie : categories) {
+            RewardCategory category = RewardCategory.valueOf(strCategorie);
+            List<Map<?, ?>> items = valuesSection.getMapList(strCategorie);
+            for (Map<?, ?> item : items) {
+                values.put(category, RewardItem.deserialize((Map<String, Object>) item));
+            }
+        }
+    }
+
+    public Collection<RewardItem> getCategoryItems(RewardCategory category) {
+        return values.get(category);
+    }
+
+    @Getter
+    @AllArgsConstructor
+    public enum RewardCategory {
+        FISH(85.0, -1.15),
+        TREASURE(5.0, 2.1),
+        TRASH(10.0, -1.95);
```

Comment:
I think `JUNK` is the proper term (but that's not really important)

Reply:
Renamed in 9fe31b6